src/Poseidon/CLI/Fetch.hs

```diff
@@ -174,17 +175,18 @@ unzipPackage zip_ outDir = do
     let archive = toArchive archiveBS
     extractFilesFromArchive [OptRecursive, OptDestination outDir] archive
 
-downloadPackage :: FilePath -> String -> String -> IO ()
-downloadPackage pathToRepo remote pacName = do
+downloadPackage :: LogEnv -> FilePath -> String -> String -> IO ()
```

Comment:
I think this function could still run in `PoseidonLogIO`, can't it? only in `printDownloadProgress` you need to switch to the manual handover of `logEnv`?

Reply:
I refactored that bit now.